core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -86,7 +86,10 @@ private[spark] object TaskDescription {
     dataOut.writeInt(taskDescription.properties.size())
     taskDescription.properties.asScala.foreach { case (key, value) =>
       dataOut.writeUTF(key)
-      dataOut.writeUTF(value)
+      // SPARK-19796 -- writeUTF doesn't work for long strings, which can happen for property values
+      val bytes = value.getBytes("utf-8")
```

Comment:
I believe in newer JDKs using the `UTF_8` constant from `StandardCharsets` is actually faster.